docs/guide/writing-tasks.md

```diff
@@ -662,6 +666,27 @@ a relation between an encrypted variable and a repository for which the encrypte
     the variable by themselves and editing both the encrypted variable and repo-owner condition
     in the `.cirrus.yml` file.
 
+### HashiCorp Vault support
+
+In addition to using Cirrus CI for managing secrets, it is possible to retrieve secrets from [HashiCorp Vault](https://www.vaultproject.io/).
+
+You will need to configure a [JWT authentication method](https://developer.hashicorp.com/vault/docs/auth/jwt#jwt-authentication) and point it to the Cirrus CI's OIDC discovery URL: `https://oidc.cirrus-ci.com`.
+
+This ensures that a cryptographic JWT token (`CIRRUS_OIDC_TOKEN`) that the Cirrus CI's task will automatically pass to your Vault will be considered valid.
```

Comment:
I think it will be good to highlight that Cirrus Agent is doing calls to vault so the Vault installation can be private.

Reply:
See https://github.com/cirruslabs/cirrus-ci-docs/pull/1088/commits/16bcc1154d28755e19169fe6a2b577866ce6c1c5.